docs/src/modules/components/AppSettingsDrawer.js

```diff
@@ -21,12 +20,6 @@ import { useTranslate } from 'docs/src/modules/utils/i18n';
 import { useChangeTheme } from 'docs/src/modules/components/ThemeContext';
 import { getCookie } from 'docs/src/modules/utils/helpers';
 
-const DrawerPaper = styled(Paper)({
-  width: 360,
-  borderTopRightRadius: 0,
-  borderBottomRightRadius: 0,
-});
-
```

Comment:
@mnajdova we might need to investigate more on this, it is very buggy and hard to reason. by using styled component and pass to `<Drawer PaperProps={{ component: DrawerPaper, square: true }}`, it is still `rounded`.

Reply:
also classname got double. `elevation-1 elevation-2`
<img width="907" alt="Screen Shot 2564-09-08 at 13 52 37" src="https://user-images.githubusercontent.com/18292247/132460644-e33bfd76-0899-4369-acdd-00662623e036.png">

Reply:
Will take a look thanks for pointing it out. I think that the general problem is that the portaled components don't support the `components` & `componentsProps` which would replace the components underneath instead of aliasing it to the `as` property (which is what we currently do).

Reply:
Currently what you have is: `<StyledPaper as={AnotherStyledPaper} />` - which results in double classes and non-deterministic classes order in the head.

Reply:
I would probably just set the `sx` prop :)

Reply:
Here is a sandbox illustrating the current behavior https://codesandbox.io/s/magical-mahavira-6p39p?file=/src/App.js